### DISCUSSION TITLE
Disable Navigation Controls

### DISCUSSION THREAD

**Discussion by oussamabonnor1** [iTwin/itwinjs-core#3246] [Q&A] [answered]
As part of our Compare experience, we have to not let the client pan/rotate/zoom the view or basically perform any action that will impact the compare between real world & model.

Is there a way to disable the listeners for mouse/keyboard events or maybe disable the tool all together on demand ?

**Comment by pmconne** [comment, marked as answer]
Any input not handled by the active tool is directed to the ["idle tool"](https://www.itwinjs.org/learning/frontend/tools/#idle-tool). The [default idle tool](https://www.itwinjs.org/reference/core-frontend/tools/idletool/) translates MMB and touch events into pan, rotate, and zoom operations.
You can replace the default idle tool with your own implementation that (for example) does nothing in response to such events by assigning to [ToolAdmin.idleTool](https://www.itwinjs.org/reference/core-frontend/tools/tooladmin/idletool/).
For some reason, the type of that property is declared as `IdleTool`, but I think it should actually be [InteractiveTool](https://www.itwinjs.org/reference/core-frontend/tools/interactivetool/). The compiler should not complain if you supply a subclass of InteractiveTool instead of IdleTool as the latter adds no additional properties/methods.

#3277 changes the type to `InteractiveTool` and adds a note to the docs regarding replacing the default idle tool.

**Reply by oussamabonnor1** [reply]
Thankn you @pmconne, you always pull through with clear detailed answers!
